openbb-agent-copilot/openbb_agent_copilot/main.py

```diff
@@ -0,0 +1,82 @@
+import re
+import json
+from pathlib import Path
+from typing import AsyncGenerator
+from openbb_agents.agent import openbb_agent
+import os
+
+from fastapi import FastAPI
+from fastapi.middleware.cors import CORSMiddleware
+from fastapi.responses import JSONResponse
+from magentic import (
+    UserMessage,
+    AssistantMessage,
+    AsyncStreamedStr,
+)
+
+from dotenv import load_dotenv
+from .models import AgentQueryRequest
+
+load_dotenv(".env")
+app = FastAPI()
+
+origins = [
+    "http://localhost",
+    "http://localhost:1420",
+    "http://localhost:5050",
+    "https://pro.openbb.dev",
+    "https://pro.openbb.co",
+]
+
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=origins,
+    allow_credentials=True,
+    allow_methods=["*"],
+    allow_headers=["*"],
+)
+
+
+def sanitize_message(message: str) -> str:
+    """Sanitize a message by escaping forbidden characters."""
+    cleaned_message = re.sub(r"(?<!\{)\{(?!{)", "{{", message)
+    cleaned_message = re.sub(r"(?<!\})\}(?!})", "}}", cleaned_message)
+    return cleaned_message
+
+
+async def create_message_stream(
+    content: AsyncStreamedStr,
+) -> AsyncGenerator[dict, None]:
+    async for chunk in content:
+        yield {"event": "copilotMessageChunk", "data": {"delta": chunk}}
+
+
+@app.get("/copilots.json")
+def get_copilot_description():
+    """Widgets configuration file for the OpenBB Terminal Pro"""
+    return JSONResponse(
+        content=json.load(open((Path(__file__).parent.resolve() / "copilots.json")))
+    )
+
+
+@app.post("/v1/query")
+async def query(request: AgentQueryRequest):
+    """Query the Copilot."""
+
+    chat_messages = []
+    for message in request.messages:
+        if message.role == "ai":
+            chat_messages.append(
+                AssistantMessage(content=sanitize_message(message.content))
+            )
+        elif message.role == "human":
+            chat_messages.append(UserMessage(content=sanitize_message(message.content)))
+
+    try:    
+        result = openbb_agent(str(chat_messages), verbose=False, openbb_pat=os.getenv("OPENBB_PAT"))
+        
+        return {"output": result}
+    
+    except Error as e:
+        error_message = e.json_body.get('error', {}).get('message', 'An unknown error occurred.')
+        return JSONResponse(status_code=400, content={"error": error_message})
```

Comment:
```suggestion
        raise HTTPException(status_code=400, content={"detail": str(e))
```

[For a bit of extra context on raising errors in FastAPI](https://fastapi.tiangolo.com/tutorial/handling-errors/).